mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -161,6 +161,33 @@ class RandomForestSuite extends SparkFunSuite with MLlibTestSparkContext {
     }
   }
 
+  test("train with empty arrays") {
+    val lp = LabeledPoint(1.0, Vectors.dense(Array.empty[Double]))
+    val data = Array.fill(5)(lp)
+    val rdd = sc.parallelize(data)
+
+    val strategy = new OldStrategy(OldAlgo.Regression, Gini, maxDepth = 2,
+      maxBins = 100)
+    intercept[IllegalArgumentException] {
+      DecisionTreeMetadata.buildMetadata(rdd, strategy)
+    }
+  }
+
+  test("train with single point") {
```

Comment:
I guess I don't see how it's an edge case _other_ than that it introduces constant features, so I'm not sure we're gaining anything here. We can leave it if you feel strongly about it.